modules/_labs/core/react/lib/typeTest.ts

```diff
@@ -0,0 +1,139 @@
+import styled from '@emotion/styled';
+import {typeColors} from '@workday/canvas-colors-web';
+import {CSSProperties, fontFamily} from '@workday/canvas-kit-react-core';
+import {default as beta_Type} from './type';
+
+export interface CanvasTypeHierarchy {
+  level10: CSSProperties;
+  level9: CSSProperties;
+  level8: CSSProperties;
+  level7: CSSProperties;
+  level6: CSSProperties;
+  level5: CSSProperties;
+  level4: CSSProperties;
+  level3: CSSProperties;
+  level2: CSSProperties;
+  level1: CSSProperties;
+  [key: string]: CSSProperties;
+}
+
+const broswerDefaultFontSize = 16;
+/**
+ * Converts px values into base 16 rems
+ */
+const rem = (pxValue: number) => {
+  return `${pxValue / broswerDefaultFontSize}rem`;
+};
+
+/**
+ * The following rem values are based on an `html` element font size of 62.5%.
+ * This takes the browser default (16px) and converts it into base 10 at the default size.
+ * For example, 1.6rem == 16px.
+ */
+
+const hierarchy: CanvasTypeHierarchy = {
+  level10: {
+    fontFamily,
+    fontSize: rem(48),
+    lineHeight: rem(56),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level9: {
+    fontFamily,
+    fontSize: rem(40),
+    lineHeight: rem(48),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level8: {
+    fontFamily,
+    fontSize: rem(32),
+    lineHeight: rem(40),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level7: {
+    fontFamily,
+    fontSize: rem(24),
+    lineHeight: rem(32),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level6: {
+    fontFamily,
+    fontSize: rem(20),
+    lineHeight: rem(28),
+    letterSpacing: rem(0.16),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level5: {
+    fontFamily,
+    fontSize: rem(18),
+    lineHeight: rem(28),
+    letterSpacing: rem(0.16),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level4: {
+    fontFamily,
+    fontSize: rem(16),
+    lineHeight: rem(24),
+    letterSpacing: rem(0.16),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+  level3: {
+    fontFamily,
+    fontSize: rem(14),
+    lineHeight: rem(20),
+    letterSpacing: rem(0.24),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+  level2: {
+    fontFamily,
+    fontSize: rem(12),
+    lineHeight: rem(16),
+    letterSpacing: rem(0.32),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+  level1: {
+    fontFamily,
+    fontSize: rem(10),
+    lineHeight: rem(16),
+    letterSpacing: rem(0.4),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+};
+
+type Variant = keyof typeof beta_Type.variant;
+type TextProps = {
+  level: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
+  variant?: Variant | Variant[];
```

Comment:
I don't know if I would allow multiple variants as you could have some odd style collisions from the [variant list](https://github.com/Workday/canvas-kit/blob/98eed9fb06d0e6b47d8e39beb15e1512471d1e73/modules/core/react/lib/type.ts#L115-L160)

Reply:
If this is something we wish to have perhaps we could expand the variations we have available... say if someone needed a mono-link variation.

Reply:
Yep, variant combinations are challenging to get right. The problem with variant arrays like this they can create unexpected results based on their order in the array. You can avoid that issue by keeping variants limited to high-level, distinct variations, and providing smaller, distinct changes through other props. Let's look at your mono + link suggestion as an example:

````tsx
<Text variant="link" fontFamily="mono">a link with mono text</Text>
````

This makes the api more declarative and reduces the chance for unintended side effects. (Though I would recommend for this particular example we create a semantic Link component that's distinct from Text.)